molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/service/impl/AlgorithmServiceImpl.java

```diff
@@ -55,13 +55,52 @@ public void autoGenerateAlgorithm(EntityMetaData sourceEntityMetaData, EntityMet
 		LOG.debug("createAttributeMappingIfOnlyOneMatch: target= " + targetAttribute.getName());
 		Iterable<AttributeMetaData> matches = semanticSearchService.findAttributes(sourceEntityMetaData,
 				targetEntityMetaData, targetAttribute);
+
 		if (Iterables.size(matches) == 1)
 		{
 			AttributeMetaData source = matches.iterator().next();
-			AttributeMapping attributeMapping = mapping.addAttributeMapping(targetAttribute.getName());
-			String algorithm = "$('" + source.getName() + "').value()";
-			attributeMapping.setAlgorithm(algorithm);
-			LOG.info("Creating attribute mapping: " + targetAttribute.getName() + " = " + algorithm);
+
+			FieldTypeEnum dataType = targetAttribute.getDataType().getEnumType();
+			String algorithm = null;
+			switch (dataType)
+			{
+				case DATE:
+				case DATE_TIME:
+					algorithm = "$('" + source.getName() + "').age().value();";
```

Comment:
What if I want to map date_time to date_time? @fdlk 
As a user might be confusing to see my date time suddenly change into age..